bench/syclblas/syclblas_benchmark_level3.cpp

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename syclblas_benchmark.cpp
+ *
+ **************************************************************************/
+
+// #include "../blas_benchmark.hpp"
+#include "../blas_benchmark2.hpp"
+
+#include <interface/blas1_interface.hpp>
+#include <interface/blas3_interface.hpp>
+
+using namespace blas;
+
+BENCHMARK_NAME_FORMAT(blas_level_3) {
+  std::ostringstream fname;
+  fname << name() << "_" << std::get<0>(params) << "_" << std::get<1>(params)
+        << "_" << std::get<2>(params) << "_" << std::get<3>(params) << "_"
+        << std::get<4>(params);
+  return fname.str();
+}
+
+BENCHMARK(gemm, blas_level_3) {
+  using ScalarT = ElemT;
+
+  size_t m = std::get<0>(params);
+  size_t k = std::get<1>(params);
+  size_t n = std::get<2>(params);
+  char const *t_a = std::get<3>(params);
+  char const *t_b = std::get<4>(params);
```

Comment:
why not std::string?

Reply:
the BLAS interface expects an old c-style string, so avoiding converting to/from string saves a little cognitive overhead.

Reply:
Ok, let it be...